src/lib/routers/history.ts

```diff
@@ -190,7 +200,11 @@ class BrowserHistory<TRouteState> implements Router<TRouteState> {
       clearTimeout(this.writeTimer);
     }
 
-    this.write({} as TRouteState);
+    if (this.writeOnDispose) {
+      this.write({} as TRouteState);
+    } else {
+      this.shouldPushState = false;
```

Comment:
What does this prevent? I've ran the tests without this line and nothing breaks.

We likely need to test this behavior if this is actually needed.